test/unit/spec/localparticipant.js

```diff
@@ -122,19 +122,16 @@ describe('LocalParticipant', () => {
     'unpublishTracks',
   ].forEach(method => {
     describe(`#${method}`, () => {
-      const trackMethod = method.slice(0, -1);
+      const trackMethod = method.match('(add|publish).*') ? 'publishTrack' : 'unpublishTrack';
 
       let test;
 
       beforeEach(() => {
         test = makeTest();
         test.participant[trackMethod] = sinon.spy(track => {
-          if (trackMethod.startsWith('publish')) {
             // TODO(mroberts): Here is a very bare-bones "mock"
             // LocalTrackPublication. We should improve this later.
             return { track, sid: track.id };
-          }
-          return track.kind === 'audio' ? null : { foo: 'bar', stop: sinon.spy() };
         });
```

Comment:
OK, so with the change above, now `test.participant[trackMethod]` is always supposed to emulate `publishTrack` or `unpublishTrack`, therefore it's OK that we return `{ track, sid: track.id }` (a little mock TrackPublication)? Also, a small nitpick: can you de-indent the body of the `if`-clause you removed?

Reply:
Yes. All the methods under test call either publishTrack or unpublishTrack. Hence the regex match above @markandrus

Reply:
Lint didnt catch indentation block. does it?